src/test/java/mate/academy/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,81 @@
+package mate.academy.service.impl;
+
+import static org.mockito.Mockito.when;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImplTest {
+    @Mock
+    private UserDao userDao;
+    private PasswordEncoder encoder;
+    private UserServiceImpl userService;
+
+    @BeforeEach
+    void setUp() {
+        encoder = new BCryptPasswordEncoder();
+        userService = new UserServiceImpl(userDao, encoder);
+    }
+
+    @Test
+    void save_Ok() {
+        User user = new User();
+        user.setPassword("1234");
+        when(userDao.save(user)).thenReturn(user);
+        Assertions.assertNotNull(userService.save(user));
+    }
+
+    @Test
+    void save_Not_Ok_NullValue() {
+        User user = new User();
+        user.setPassword("1234");
+        Assertions.assertNull(userService.save(user));
+    }
```

Comment:
This and similar cases are not valid tests.
You dao returning null just because it's mocked without prescribed behaviour. 
Or if your code/db constraints are so bad that when you pass user without login then your dao saves it without any warnings and returns null (can't imagine even how this could be possible).

99% that your dao will throw DataProcessingException on such request from service. So we should do the same in test:
```suggestion
    @Test
    void save_Dao_Produced_Exception_Not_Ok() {
        User user = new User();
        user.setPassword("1234");
        when(userDao.save(any()).thenThrow(new DataProcessingException("some message"));
        Assertions.assertThrows(DataProcessingException.class, () -> userService.save(user));
    }
```

But I don't think it makes any sense to write such test as it's obvious that runtime exception throw in child method will appear in parent one as we do not catch it.
So probably better to just remove such tests.

And please recheck all your "Null" test cases. I don't think that it normal behaviour for any service or dao method to return null